test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -37,11 +37,39 @@
 
 from torch.nn.utils.rnn import PackedSequence
 
-def to_numpy(tensor):
-    if tensor.requires_grad:
-        return tensor.detach().cpu().numpy()
+
+def flatten_tuples(elem):
+    tup = []
+    for t in elem:
+        if isinstance(t, (tuple)):
+            tup += flatten_tuples(t)
+        else:
+            tup += [t]
+    return tup
+
+
+def to_numpy(elem):
+    if isinstance(elem, torch.Tensor):
+        if elem.requires_grad:
+            return elem.detach().cpu().numpy()
+        else:
+            return elem.cpu().numpy()
+    elif isinstance(elem, list) or isinstance(elem, tuple):
+        return [to_numpy(inp) for inp in elem]
+    elif isinstance(elem, bool):
+        return np.array(elem, dtype=bool)
+    elif isinstance(elem, int):
+        return np.array(elem, dtype=int)
+    elif isinstance(elem, float):
+        return np.array(elem, dtype=float)
+    elif isinstance(elem, dict):
+        dict_ = []
+        for k in elem:
+            dict_ += [to_numpy(k)] + [to_numpy(elem[k])]
```

Comment:
Are we flattening both the key and value in the case of a dict?

Reply:
Yes, keys and values are flattened and concatenated.

Reply:
how does it work for dict? is there any current test case for it?

Reply:
So far we only test dict when tracing, and for those cases torch jit faltten pass is used to flatten input/outputs.
In those cases, the jit flatten pass creates a list of flattened keys and values of the dict.
But no, we don't have a test for actual Dict input/outputs (scripting) as those are not supported by ONNX. This part is added for completeness.